napari/layers/utils/_slice_input.py

```diff
@@ -78,17 +106,41 @@ def data_indices(
             )
 
         slice_world_to_data = world_to_data.set_slice(self.not_displayed)
-        world_pts = [self.point[ax] for ax in self.not_displayed]
+        world_pts = [self.world_slice.point[ax] for ax in self.not_displayed]
+
+        world_margin_left = [
+            self.world_slice.point[ax] - self.world_slice.margin_left[ax]
+            for ax in self.not_displayed
+        ]
+        world_margin_right = [
+            self.world_slice.point[ax] + self.world_slice.margin_right[ax]
+            for ax in self.not_displayed
+        ]
+
         data_pts = slice_world_to_data(world_pts)
+        data_margin_left = slice_world_to_data(world_margin_left)
+        data_margin_right = slice_world_to_data(world_margin_right)
+
         if round_index:
             # A round is taken to convert these values to slicing integers
             data_pts = np.round(data_pts).astype(int)
+            data_margin_left = np.round(data_margin_left).astype(int)
+            data_margin_right = np.round(data_margin_right).astype(int)
 
-        indices = [slice(None)] * self.ndim
-        for i, ax in enumerate(self.not_displayed):
-            indices[ax] = data_pts[i]
+        full_pts = [np.nan for ax in range(self.ndim)]
+        full_margin_left = [np.nan for _ in range(self.ndim)]
+        full_margin_right = [np.nan for _ in range(self.ndim)]
 
-        return tuple(indices)
+        for i, ax in enumerate(self.not_displayed):
+            full_pts[ax] = data_pts[i]
+            full_margin_left[ax] = data_margin_left[i]
+            full_margin_right[ax] = data_margin_right[i]
+
+        return _ThickNDSlice(
+            point=full_pts,
+            margin_left=full_margin_left,
+            margin_right=full_margin_right,
```

Comment:
Based on the typing of `_ThickNDSlice`, I think this needs to be:

```suggestion
            point=tuple(full_pts),
            margin_left=tuple(full_margin_left),
            margin_right=tuple(full_margin_right),
```

Changing the typing in the class is also fine with me, but i do love immutable things :)